agent/check_handler.go

```diff
@@ -223,3 +224,44 @@ func extractMetrics(event *types.Event) []*types.MetricPoint {
 
 	return transformer.Transform()
 }
+
+// mergeEnvironments merges env2 into env1, overwriting any existing variable
+// in env1, except for the "special" variables PATH, CPATH and LD_LIBRARY_PATH.
+// These variables are merged by prepending the value from env2 to the value in
+// env1, effectively giving priority to the value from env2.
+func mergeEnvironments(env1, env2 []string) []string {
+	toMap := func(s []string) map[string]string {
+		m := map[string]string{}
+
+		for _, v := range s {
+			split := strings.SplitN(v, "=", 2)
```

Comment:
SplitN with an `n` parameter of 2 does not make sense here, as we expect to find only a single `=`. Also, we should probably do a sanity check here that we actually get a slice with len(s) > 1, as not to inadvertently panic on invalid environment variables.

Reply:
I disagree, we do not expect to find a single `=`. An environment variable's value is an arbitrary string. Here is a counter example:

```
MY_SUPER_IMPORTANT_VAR="launch_missile=false && eat_tacos=true"
```

All I think we can expect is that the left side of the _first_ `=` is a variable name (the key in the map) and its left side is the associated value.

I agree with checking the size of `split` though!

Reply:
But shouldn't it at least be SplitN with a parameter of `1` then?